services/dialog/v1.js

```diff
@@ -0,0 +1,342 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+
+  var request = require("request");
+  var cfenv = require("cfenv");
+  var fs = require("fs");
+  var temp = require("temp");
+  var async = require("async");
+  var watson = require("watson-developer-cloud");
+  temp.track();
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then they are stored in sUsername and sPassword, as the 
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+	
+  var services = cfenv.getAppEnv().services;
+
+  var username, password, sUsername, sPassword;
+  var service = cfenv.getAppEnv().getServiceCreds(/dialog/i);
+  
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+  
+  RED.httpAdmin.get("/service-dialog/vcap", function (req, res) {
+		res.json(service ? {bound_service: true} : null);
+  });  
+  
+  
+  // This is the Dialog Node. 
+  // The node supports six modes
+  // create a dialog from a dialog template
+  // delete a dialog : the Watson Dialog service Instance is limited to 10 dialog instances
+  // list - retrieves a list of the Dialogs
+  // start conversation - initiates a conversation with an existing conversation
+  // continue a conversation - continues with a started conversation
+  // get profile variable - GETs the profile variables associated with the dialog
+
+  function WatsonDialogNode (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+   
+    this.log("Watson Developer Cloud Contribution - Dialog Node Instantiated") ;
+	
+    this.on("input", function (msg) {
+      this.log("Watson Developer Cloud Contribution - Input received");
+		
+      var message = "";
+      if (!msg.payload) {
+			 message = "Missing property: msg.payload";
+			 node.error(message, msg);
+			 return;
+      }
+
+      var username = sUsername || this.credentials.username;
+      var password = sPassword || this.credentials.password;
+      this.status({});  
+			
+      if (!username || !password) {
+        this.status({fill:"red", shape:"ring", text:"missing credentials"});  
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 69.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2385269578/issues/source?bid=3225115&fileBranchId=3225116#l80)